### PR TITLE
Use arrow functions

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Issue: https://github.com/pdepend/pdepend/wiki/3.0-Roadmap
Breaking change: no

### CODE REVIEW

File: scripts/php-keywords.php [hunk 1]
```diff
@@ -126,9 +126,7 @@ protected function is%sName($tokenType)
         join(
             "\n",
             array_map(
-                function ($token) {
-                    return sprintf('            case Tokens::%s:', $token);
-                },
+                fn ($token) => sprintf('            case Tokens::%s:', $token),
```
Comment (kylekatarnls) [inline scripts/php-keywords.php:129]:
```suggestion
                static fn ($token) => sprintf('            case Tokens::%s:', $token),
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 2]
```diff
@@ -249,9 +249,7 @@ private function getItems()
         // Sort items by size
         usort(
             $items,
-            function ($a, $b) {
-                return ($a['size'] - $b['size']);
-            },
+            fn ($a, $b) => ($a['size'] - $b['size']),
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:252]:
```suggestion
            static fn (array $a, array $b) => $a['size'] <=> $b['size'],
```

File: src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php [hunk 3]
```diff
@@ -7838,9 +7838,7 @@ private function parseVarAnnotation($comment)
             $useSymbolTable = $this->useSymbolTable;
 
             return array_map(
-                function ($image) use ($useSymbolTable) {
-                    return $useSymbolTable->lookup($image) ?: $image;
-                },
+                fn ($image) => $useSymbolTable->lookup($image) ?: $image,
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:7841]:
```suggestion
                static fn ($image) => $useSymbolTable->lookup($image) ?: $image,
```

File: src/test/php/PDepend/Bugs/Phpmd/FunctionDocBlockBugPhpmd914Test.php [hunk 4]
```diff
@@ -56,9 +56,7 @@ public function testFunctionDocBlockCanBeRead()
         /** @var ASTFunction $function */
         $function = $this->getFirstFunctionForTestCase();
 
-        $lines = array_map(function ($line) {
-            return trim($line, " \t\n\r\0\x0B/*");
-        }, explode("\n", trim($function->getComment(), " \t\n\r\0\x0B/*")));
+        $lines = array_map(fn ($line) => trim($line, " \t\n\r\0\x0B/*"), explode("\n", trim($function->getComment(), " \t\n\r\0\x0B/*")));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Bugs/Phpmd/FunctionDocBlockBugPhpmd914Test.php:59]:
```suggestion
        $lines = array_map(
            static fn (string $line) => trim($line, " \t\n\r\0\x0B/*"),
            explode("\n", trim($function->getComment(), " \t\n\r\0\x0B/*")),
        );
```

File: src/test/php/PDepend/Issues/ReflectionCompatibilityIssue067Test.php [hunk 5]
```diff
@@ -379,9 +379,7 @@ public function testParserHandlesParameterDefaultValueClassConstant()
         $this->assertSame('\\PDepend\\Code', $parameters[0]->getDefaultValue()->getImage());
         /** @var ASTNode $node */
         $node = $parameters[0]->getDefaultValue()->getChild(0);
-        $image = implode($node->getImage(), array_map(function (ASTNode $node) {
-            return $node->getImage();
-        }, $node->getChildren()));
+        $image = implode($node->getImage(), array_map(fn (ASTNode $node) => $node->getImage(), $node->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Issues/ReflectionCompatibilityIssue067Test.php:382]:
```suggestion
        $image = implode(
            $node->getImage(),
            array_map(static fn (ASTNode $node) => $node->getImage(), $node->getChildren()),
        );
```

File: src/test/php/PDepend/Metrics/Analyzer/ClassDependencyAnalyzerTest.php [hunk 6]
```diff
@@ -120,15 +120,11 @@ public function testGenerateMetrics()
         foreach ($namespaces as $namespace) {
             foreach ($namespace->getTypes() as $type) {
                 $actual[$type->getName()]['efferent'] = array_map(
-                    function ($type) {
-                        return $type->getName();
-                    },
+                    fn ($type) => $type->getName(),
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Metrics/Analyzer/ClassDependencyAnalyzerTest.php:123]:
```suggestion
                    static fn ($type) => $type->getName(),
```

File: src/test/php/PDepend/Metrics/Analyzer/ClassDependencyAnalyzerTest.php [hunk 7]
```diff
@@ -120,15 +120,11 @@ public function testGenerateMetrics()
...
                     $visitor->getEfferents($type)
                 );
                 $actual[$type->getName()]['afferent'] = array_map(
-                    function ($type) {
-                        return $type->getName();
-                    },
+                    fn ($type) => $type->getName(),
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Metrics/Analyzer/ClassDependencyAnalyzerTest.php:127]:
```suggestion
                    static fn ($type) => $type->getName(),
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php [hunk 8]
```diff
@@ -137,16 +137,12 @@ private function checkMatchExpression($namespacePrefix = null)
...
-            array('PDepend\\Source\\AST\\ASTLiteral', ']'),
-        ), array_map(function ($node) {
-            return array(get_class($node), $node->getImage());
-        }, $new->getChild(1)->getChild(0)->getChildren()));
+        $this->assertSame(['\InvalidArgumentException', ''], array_map(fn ($node) => $node->getImage(), $new->getChildren()));
+        $this->assertSame([
+            ['PDepend\\Source\\AST\\ASTLiteral', 'Invalid code ['],
+            ['PDepend\\Source\\AST\\ASTVariable', '$in'],
+            ['PDepend\\Source\\AST\\ASTLiteral', ']'],
+        ], array_map(fn ($node) => [$node::class, $node->getImage()], $new->getChild(1)->getChild(0)->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php:145]:
```suggestion
        ], array_map(
            static fn ($node) => [$node::class, $node->getImage()],
            $new->getChild(1)->getChild(0)->getChildren(),
        ));
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php [hunk 9]
```diff
@@ -215,16 +211,12 @@ public function testMatchExpressionWithMultipleKeyExpressions()
...
-            return $node->getImage();
-        }, $new->getChildren()));
-        $this->assertSame(array(
-            array('PDepend\\Source\\AST\\ASTLiteral', 'Invalid code ['),
-            array('PDepend\\Source\\AST\\ASTVariable', '$in'),
-            array('PDepend\\Source\\AST\\ASTLiteral', ']'),
-        ), array_map(function ($node) {
-            return array(get_class($node), $node->getImage());
-        }, $new->getChild(1)->getChild(0)->getChildren()));
+        $this->assertSame(['\InvalidArgumentException', ''], array_map(fn ($node) => $node->getImage(), $new->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php:214]:
```suggestion
        $this->assertSame(['\InvalidArgumentException', ''], array_map(
            static fn ($node) => $node->getImage(),
            $new->getChildren(),
        ));
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php [hunk 10]
```diff
@@ -215,16 +211,12 @@ public function testMatchExpressionWithMultipleKeyExpressions()
...
+        $this->assertSame([
+            ['PDepend\\Source\\AST\\ASTLiteral', 'Invalid code ['],
+            ['PDepend\\Source\\AST\\ASTVariable', '$in'],
+            ['PDepend\\Source\\AST\\ASTLiteral', ']'],
+        ], array_map(fn ($node) => [$node::class, $node->getImage()], $new->getChild(1)->getChild(0)->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP80/MatchExpressionTest.php:219]:
```suggestion
        ], array_map(
            static fn ($node) => [$node::class, $node->getImage()],
            $new->getChild(1)->getChild(0)->getChildren(),
        ));
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php [hunk 11]
```diff
@@ -108,9 +108,7 @@ public function testEnum()
                 'HasColor' => 'HasColor',
             ),
             array_map(
-                function (ASTInterface $interface) {
-                    return $interface->getName();
-                },
+                fn (ASTInterface $interface) => $interface->getName(),
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php:111]:
```suggestion
                static fn (ASTInterface $interface) => $interface->getName(),
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php [hunk 12]
```diff
@@ -122,9 +120,7 @@ function (ASTInterface $interface) {
                 'getcolor' => 'getColor',
             ),
             array_map(
-                function (ASTMethod $interface) {
-                    return $interface->getName();
-                },
+                fn (ASTMethod $interface) => $interface->getName(),
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php:123]:
```suggestion
                static fn (ASTMethod $interface) => $interface->getName(),
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php [hunk 13]
```diff
@@ -138,9 +134,7 @@ function (ASTMethod $interface) {
                 'SPADES' => "'spades'",
             ),
             array_map(
-                function (ASTEnumCase $case) {
-                    return $case->getValue()->getImage();
-                },
+                fn (ASTEnumCase $case) => $case->getValue()->getImage(),
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP81/EnumTest.php:137]:
```suggestion
                static fn (ASTEnumCase $case) => $case->getValue()->getImage(),
```

File: src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php [hunk 14]
```diff
@@ -207,9 +207,7 @@ public function testArrowFunctions()
             '$number',
             '*',
             '2',
-        ), array_map(function (ASTNode $node) {
-            return $node->getImage();
-        }, $expression->getChildren()));
+        ), array_map(fn (ASTNode $node) => $node->getImage(), $expression->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php:210]:
```suggestion
        ), array_map(
            static fn (ASTNode $node) => $node->getImage(),
            $expression->getChildren(),
        ));
```

File: src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php [hunk 15]
```diff
@@ -253,9 +251,7 @@ public function testArrowFunctionsWithReturnType()
...
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php:254]:
```suggestion
        ), array_map(
            static fn (ASTNode $node) => $node->getImage(),
            $expression->getChildren(),
        ));
```

File: src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php [hunk 16]
```diff
@@ -288,9 +284,7 @@ public function testUnpackingInsideArrays()
             'PDepend\\Source\\AST\\ASTArrayElement',
         ), array_map('get_class', $expression->getChildren()));
         /** @var ASTNode[] $elements */
-        $elements = array_map(function ($node) {
-            return $node->getChild(0);
-        }, $expression->getChildren());
+        $elements = array_map(fn ($node) => $node->getChild(0), $expression->getChildren());
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php:287]:
```suggestion
        $elements = array_map(
            static fn ($node) => $node->getChild(0),
            $expression->getChildren(),
        );
```

File: src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php [hunk 17]
```diff
@@ -303,9 +297,7 @@ public function testUnpackingInsideArrays()
         $this->assertSame(array(
             '...',
             '$numbers',
-        ), array_map(function (ASTNode $node) {
-            return $node->getImage();
-        }, $expression->getChildren()));
+        ), array_map(fn (ASTNode $node) => $node->getImage(), $expression->getChildren()));
```
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/PHPParserVersion74Test.php:300]:
```suggestion
        ), array_map(
            static fn (ASTNode $node) => $node->getImage(),
            $expression->getChildren(),
        ));
```

**Review by kylekatarnls — DISMISSED** [review]
Approved, but optionally none of them use `$this` so they could be static.

File: src/main/php/PDepend/Report/Jdepend/Chart.php (hunk 2 above)
Comment (AJenbo) [reply src/main/php/PDepend/Report/Jdepend/Chart.php:252]: Nice, I'm a fan of the space ship :)

**Comment by AJenbo** [comment]
Thanks, looks nicer with the line breaks you added as well.

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**